framework/db/BaseActiveRecord.php

```diff
@@ -409,6 +410,43 @@ public function hasMany($class, $link)
     }
 
     /**
+     * Declares a relation from the another class relation definition use `via()`.
+     *
+     * For example, to declare the `items` relation for `Order` class, we can write
+     * the following code in the `Order` class:
+     *
+     * ```php
+     * public function getItems(){
+     *     return $this->hasRelation(OrderItem::className(), 'item')->via('orderItems');
+     * }
+     * public function getOrderItems()
+     * {
+     *     return $this->hasMany(OrderItem::className(), ['order_id' => 'id']);
+     * }
+     * ```
+     *
+     * and we can write following code in the `OrderItem` class:
+     *
+     * ```php
+     * public function getItem()
+     * {
+     *     return $this->hasOne(Item::className(), ['id' => 'item_id']);
+     * }
+     * ```
+     *
+     * @param ActiveRecord|array|callable|string $target the model class who have relation
+     * @param string $relationName relation name in target (case sensitive, without 'get')
+     * @return ActiveQuery
+     */
+    public function hasRelation($target, $relationName){
+        $target instanceof ActiveRecord || $target = Yii::createObject($target);
```

Comment:
like a hasOne, hasMany?

So... How about `fromRelation` or `viaRelation` (and directly add via into method)?